pkg/globalnet/controllers/uninstall.go

```diff
@@ -216,3 +218,32 @@ func RemoveGlobalIPAnnotationOnNode(cfg *rest.Config) {
 
 	logger.Infof("Successfully removed globalIP annotation from node %q", nodeName)
 }
+
+func RemoveStaleInternalServices(config *syncer.ResourceSyncerConfig) error {
+	_, gvr, err := util.ToUnstructuredResource(&corev1.Service{}, config.RestMapper)
+	if err != nil {
+		return errors.Wrap(err, "error converting resource")
+	}
+
+	services := config.SourceClient.Resource(*gvr)
+
+	svcList, err := services.Namespace(corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error listing the services")
+	}
+
+	for i := range svcList.Items {
+		svc := &svcList.Items[i]
+		if !svc.GetDeletionTimestamp().IsZero() {
+			if finalizer.IsPresent(svc, InternalServiceFinalizer) {
+				err := finalizer.Remove(context.TODO(), resource.ForDynamic(services.Namespace(svc.GetNamespace())), svc,
+					InternalServiceFinalizer)
+
+				return errors.Wrapf(err, "error while removing the finalizer from globalnet internal service %q/%q",
+					svc.GetNamespace(), svc.GetName())
```

Comment:
My prior suggestion was incorrect - I guess I misread the code - sorry about that. It should be the way you had it.

```suggestion
				if err != nil {
				    return errors.Wrapf(err, "error while removing the finalizer from globalnet internal service %q/%q",
					    svc.GetNamespace(), svc.GetName())
				}
```